### PR TITLE
libmad: update to 0.16.2

### PR DESCRIPTION
**Pull request by neheb**
Backport aarch64 patch.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @thess 
Compile tested: mips24

@Vladdrako please test

### CODE REVIEW

**Comment by Vladdrako** [comment]
Sadly, the issue is present.

**Comment by neheb** [comment]
Guess I'll ping @@Be-ing

**Comment by Be-ing** [comment]
What is the issue? I don't see a failure on the CI for this PR.

**Comment by neheb** [comment]
crackling sound under MIPS.

**Comment by neheb** [comment]
I tried to look into this. One difference I see is:

```
mangix@ryzen ~/d/openwrt (mangix)> strings build_dir/target-mipsel_24kc_musl/libmad-0.15.1b/ipkg-install/usr/lib/libmad.
so.0.2.1 | grep MIPS
NDEBUG FPM_MIPS ASO_INTERLEAVE2 ASO_ZEROCHECK OPT_SPEED OPT_SSO
_MIPS_STUBS_
.MIPS.abiflags
.MIPS.stubs
mangix@ryzen ~/d/openwrt (mangix)> strings build_dir/target-mips64_mips64r2_64_musl/libmad-0.16.2/ipkg-install/usr/lib/l
ibmad.so.0.16.2 | grep MIPS
NDEBUG FPM_MIPS ASO_INTERLEAVE2 ASO_ZEROCHECK OPT_ACCURACY
```

Moved back to SPEED.

@Vladdrako please retest.

**Comment by Vladdrako** [comment]
Works fine now, thanks 👍🏻

File: libs/libmad/Makefile [hunk 1]
```diff
@@ -39,6 +40,11 @@ define Package/libmad/description
 	Layer I, Layer II, and Layer III (i.e. MP3) - are fully implemented.
 endef
 
+CMAKE_OPTIONS += \
+	-DOPTIMIZE=SPEED
+
+TARGET_CFLAGS:= $(filter-out -O%,$(TARGET_CFLAGS)) -O2
```
Comment (Be-ing) [inline libs/libmad/Makefile:46]: FWIW CMake sets `-O3` by default for `CMAKE_BUILD_TYPE=Release`

**Pull request merged**